zrml/neo-swaps/src/lib.rs

```diff
@@ -571,13 +805,18 @@ mod pallet {
                         >= LN_NUMERICAL_LIMIT.saturated_into(),
                     Error::<T>::NumericalLimits(NumericalLimitsError::MinAmountNotMet),
                 );
+                let buy = vec![asset_out];
+                let sell = pool.assets_complement(&buy);
                 let swap_amount_out =
-                    pool.calculate_swap_amount_out_for_buy(asset_out, amount_in_minus_fees)?;
+                    pool.calculate_swap_amount_out_for_buy(buy, sell, amount_in_minus_fees)?;
                 let amount_out = swap_amount_out.checked_add_res(&amount_in_minus_fees)?;
```

Comment:
Maybe it's worth commenting here that `swap_amount_out` is just `y(x)`, means the amount of assets in `sell` (`S`) that were sold for more assets of `buy` (`B`). That's why `amount_out = swap_amount_out + amount_in_minus_fees`, which is the buy complete set amount plus the additional amount that was received through the sale of the unwanted outcomes in `sell`.

Reply:
cf6a9d27de3a62e70b57278ce71637b76315a58a